src/java/org/apache/cassandra/cql3/ResultSet.java

```diff
@@ -316,6 +317,8 @@ public static ResultMetadata fromPrepared(CQLStatement statement)
         {
             if (statement instanceof SelectStatement)
                 return ((SelectStatement)statement).getResultMetadata();
+            if (statement instanceof TransactionStatement)
```

Comment:
makes me wonder if this should be an interface to remove duplication

Reply:
Yup, `ComplexStatement` to the rescue.

Reply:
testing out

```
public interface ReturningCQLStatement extends CQLStatement
{
    ResultSet.ResultMetadata getResultMetadata();
}
```